tasks/install_configure_peertube.yml

```diff
@@ -0,0 +1,119 @@
+    group: '{{ peertube_group }}'
+    mode: 0755
+  loop:
+    - config
+    - storage
+    - versions
+
+- stat: path={{ peertube_user_path }}/versions/peertube-{{ peertube_version }}
+  register: peertube_version_dir
+
+- name: Download and extract Peertube
+  unarchive:
+    src: "https://github.com/Chocobozzz/PeerTube/releases/download/{{ peertube_version }}/peertube-{{ peertube_version }}.zip"
+    dest: "{{ peertube_user_path }}/versions"
+    remote_src: yes
+    owner: '{{ peertube_user }}'
+    group: '{{ peertube_group }}'
+  when: peertube_version_dir.stat.exists == False
+  register: fresh_install
+
+- stat: path={{ peertube_user_path }}/peertube-latest
+  register: peertube_latest_dir
+
+- name: Create symlink peertube-latest
+  file:
+    src: "{{ peertube_user_path }}/versions/peertube-{{ peertube_version }}"
+    dest: "{{ peertube_user_path }}/peertube-latest"
+    state: link
+    owner: '{{ peertube_user }}'
+    group: '{{ peertube_group }}'
+  when: peertube_latest_dir.stat.exists == False
+
+- name: Change owner of all peertube directories # seems setting owner on unarchive is not enough…
+  file:
+    path: "{{ peertube_user_path }}"
+    owner: '{{ peertube_user }}'
+    group: '{{ peertube_group }}'
+    recurse: yes
+
+- name: Copy configuration peertube
+  template:
+    src: "production.yaml.j2"
+    dest: '{{ peertube_user_path }}/config/production.yaml'
+    owner: '{{ peertube_user }}'
+    group: '{{ peertube_group }}'
+    mode: 0644
+  notify:
+    - restart peertube daemon
+
+- name: Install Stretch backports repo
+  apt_repository:
+    repo: deb http://ftp.debian.org/debian stretch-backports main
+    state: present
+  when:
+    - (ansible_distribution_major_version == "9")
+  register: backports_installed
+
+- name: Change apt_preference
+  copy:
+    content: 'APT::Default-Release "stable";'
+    dest: /etc/apt/apt.conf.d/99apt_default
+    mode: 644
+  when: backports_installed is changed
+
+- name: Install Peertube daemon
+  template:
+    src: peertube.service
+    dest: /etc/systemd/system/
+  notify:
+    - install peertube daemon
+
+- name: Enabled service Peertube
+  service:
+    name: peertube
+    enabled: yes
+  notify: install peertube daemon
+
+- name: Install plugins npm package
+  shell: cd {{ peertube_user_path }}/peertube-latest && sudo -u peertube NODE_CONFIG_DIR={{ peertube_user_path }}/config NODE_ENV=production npm run plugin:install -- --npm-name {{ item }}
+  loop: "{{ peertube_plugins }}"
+
+- name: Copy additional SQL queries
+  template:
+    src: sql/{{ item }}
+    dest: /tmp/{{ item }}
+    owner: '{{ peertube_user }}'
+    group: '{{ peertube_group }}'
+    mode: 0755
+  loop: "{{ peertube_sql_extras }}"
+
+- name: Run additional SQL queries
+  become: yes
+  become_user: '{{ peertube_postgres_user }}'
+  command: psql -d peertube_prod -U postgres -f '/tmp/{{ item }}'
+  loop: "{{ peertube_sql_extras }}"
+  notify: restart peertube daemon
+
+- name: Update dependencies
```

Comment:
To keep the logic of the original role we can add a `when: fresh_install is changed` here.

Reply:
Added [here](https://github.com/UdelaRInterior/ansible-peertube/blob/724374451673d1f91aac02dd9afa9255acaa2c9e/tasks/install_configure_peertube.yml#L101)